gaffer-as-a-service/gaas-rest/src/main/java/uk/gov/gchq/gaffer/gaas/factories/GafferSpecBuilder.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright 2020 Crown Copyright
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.gchq.gaffer.gaas.factories;
+
+import uk.gov.gchq.gaffer.common.model.v1.GafferSpec;
+import static uk.gov.gchq.gaffer.gaas.util.Constants.DESCRIPTION_KEY;
+import static uk.gov.gchq.gaffer.gaas.util.Constants.GRAPH_ID_KEY;
+import static uk.gov.gchq.gaffer.gaas.util.Constants.INGRESS_API_PATH_KEY;
+import static uk.gov.gchq.gaffer.gaas.util.Constants.INGRESS_HOST_KEY;
+import static uk.gov.gchq.gaffer.gaas.util.Constants.INGRESS_UI_PATH_KEY;
+import static uk.gov.gchq.gaffer.gaas.util.Properties.INGRESS_SUFFIX;
+import static uk.gov.gchq.gaffer.gaas.util.Properties.NAMESPACE;
+public class GafferSpecBuilder {
```

Comment:
Not sure if you need this `class`... Seems you could just build a `GafferSpec` object in the `AbstractStoreTypeBuilder.build()` instead and just store the `variables`.

Reply:
Sorry, this goes against my sample code found here (https://github.com/gchq/gaffer-experimental/issues/79) but it would mean fewer classes.

Reply:
Done